resources/qml/PrinterOutput/ManualPrinterControl.qml

```diff
@@ -429,6 +427,120 @@ Item
             }
         }
 
+        Row
+        {
+            id: customCommandInputRow
+
+            width: base.width - 2 * UM.Theme.getSize("default_margin").width
+            height: childrenRect.height + UM.Theme.getSize("default_margin").width
+            anchors.left: parent.left
+            anchors.leftMargin: UM.Theme.getSize("default_margin").width
+
+            spacing: UM.Theme.getSize("default_margin").width
+
+            Label
+            {
+                text: catalog.i18nc("@label", "Send G-code")
+                color: UM.Theme.getColor("setting_control_text")
+                font: UM.Theme.getFont("default")
+
+                width: Math.floor(parent.width * 0.4) - UM.Theme.getSize("default_margin").width
+                height: UM.Theme.getSize("setting_control").height
+                verticalAlignment: Text.AlignVCenter
+            }
+
+            Row
+            {
+                // Input field for custom G-code commands.
+                Rectangle
+                {
+                    id: customCommandControl
+
+                    // state
+                    visible: printerModel != null ? printerModel.canSendRawGcode: true
+                    enabled: {
+                        if (printerModel == null) {
+                            return false // Can't preheat if not connected
+                        }
+                        if (!connectedPrinter.acceptsCommands) {
+                            return false // Not allowed to do anything
+                        }
+                        if (connectedPrinter.jobState == "printing" || connectedPrinter.jobState == "pre_print" || connectedPrinter.jobState == "resuming" || connectedPrinter.jobState == "pausing" || connectedPrinter.jobState == "paused" || connectedPrinter.jobState == "error" || connectedPrinter.jobState == "offline") {
+                            return false // Printer is in a state where it can't react to pre-heating
```

Comment:
Doesn't matter if it can react to pre-heating or not, though.

It's probably wise to put this code somewhere in a property at the top of the file. We currently repeat this check at several places (case in point, you copy-pasted it from the pre-heating code). Giving the root item a check like this can prevent that:

    property var notIdling: connectedPrinter.jobState == "printing" || connectedPrinter.jobState == "pre_print" || connectedPrinter.jobState == "resuming" || connectedPrinter.jobState == "pausing" || connectedPrinter.jobState == "paused" || connectedPrinter.jobState == "error" || connectedPrinter.jobState == "offline"

(or maybe something more semantic like `connectedPrinter.jobState != "idle"`)

And then use `if (notIdling)` instead of this check here and at line 123.

Reply:
In the interest of development speed I think I'll just make this change myself for you to help out. It's just a style thing after all.